snapshot/snapshotfs/upload.go

```diff
@@ -606,6 +611,12 @@ func uniqueDirectories(dirs []fs.Directory) []fs.Directory {
 	return result
 }
 
+// DirReadError distinguishes an error thrown when attempting to
+// read a directory
+type DirReadError struct {
```

Comment:
Does `DirReadError` need to be exported?

Reply:
you're right, probably not

Reply:
unexported